pilot/pkg/config/kube/ingress/status.go

```diff
@@ -84,12 +80,13 @@ func NewStatusSyncer(meshHolder mesh.Holder, client kubelib.Client) *StatusSynce
 		nodeLister:         client.KubeInformer().Core().V1().Nodes().Lister(),
 		ingressClassLister: ingressClassLister,
 		queue:              q,
-		ingressService:     meshHolder.Mesh().IngressService,
-		ingressSelector:    meshHolder.Mesh().IngressSelector,
 	}
 }
 
 func (s *StatusSyncer) onEvent() error {
+	if !s.ingressGatewayExists() {
```

Comment:
nit: rather than an extra GET should we just check the resulting err from runningAddresses for IsNotFound? Just to avoid extra calls

Reply:
Makes sense, done